src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -238,6 +250,9 @@ CHIP_ERROR DeviceControllerFactory::InitSystemState(FactoryInitParams params)
     ReturnErrorOnFailure(stateParams.unsolicitedStatusHandler->Init(stateParams.exchangeMgr));
     ReturnErrorOnFailure(stateParams.bdxTransferServer->Init(stateParams.systemLayer, stateParams.exchangeMgr));
 
+    chip::app::InteractionModelEngine * interactionModelEngine = chip::app::InteractionModelEngine::GetInstance();
+    interactionModelEngine->SetDataModelProvider(params.dataModelProvider);
```

Comment:
Please document that this must come before InitDataModelHandler at the moment and why.